### PR TITLE
[diff] Miscellaneous fixes.

### LINKED ISSUE #214: Many fixes in [diff]
From an email to cxxeditor@ from Guilliam Xavier:

Hello Mr. Du Toit,

While reading Annex C (informative) Compatibility [diff] in N3690, I noticed some typos and other problems. One of them has been recently fixed in the Git repository (wrong C headers count) but there are more. I don't have a GitHub account for now so here's an email.

First, I am attaching some editorial fixes. The modifications are all located in the source file "compatibility.tex" (current version, last change to the file on GitHub draft repository was commit a7e0895c869adba3b091607926e31faff9f8e9cf "[diff.mods.to.headers] Remove redundant (and wrong) count of C headers." of Sep 26, 2013).
#1.diff

Editorial fixes that I think should be applied.
- [diff.basic] [diff.expr] [diff.dcl] Add an effective line break (4 locations).
- [diff.basic] Delete spurious \rationale macro in \difficulty text.
- [diff.basic] Insert missing space between words.
- [diff.basic] [diff.cpp03.conv] Add a final period for consistency (5 locations).
- [diff.expr] Remove "a" article for consistency.
- [diff.dcl] [diff.class] Replace "C++" with the \Cpp macro (2 locations).
- [diff.class] Fix typo "choise" -> "choice".
- [diff.class] Replace "Change of semantics of" with "Change to semantics of" for consistency.
- [diff.special] Make example really "valid in ISO C" by using copy-initialization syntax.
- [diff.cpp03.lex] Fix typo "tokens" -> "token" ("a single preprocessing token").
- [diff.cpp03.library] Fix typo "in This" -> "in this" (lowercase T).
- [diff.cpp03.containers] Fix wrong macro use \effects -> \effect.
#2.diff

Editorial fixes that I am less sure about.
- [diff.basic] Format "Main" in code font ("main").
- [diff.expr] Remove extra space before period.
- [diff.dcl] Format "const" in code font.

(For that last one, an alternative would be to capitalize "const" to "Const".)
#3.diff

Editorial fix for consistency. Among the 67 "Change:" headings, 58 don't end the first line with a period, but 9 do; that feels somewhat inconsistent. As for whether add 58 periods or remove 9, the second alternative is a smaller change.
- [diff.dcl] [diff.decl] [diff.class] [diff.cpp03.special] [diff.cpp11.dcl.dcl] Remove final period for consistency (9 locations).

Note: when trying to apply the patches, I saw that 2.diff can only be applied after 1.diff has been applied.

Feel free to tweak the modifications.

Additionally, in [diff.library], paragraph 4 reads "The C++ standard library provides 57 standard values from the C library, as shown in Table 150." but Table 150 shows 45 values, not 57 (Table 149 does show 57 standard macros, so that looks like a problem of copy-paste from paragraph 2).
But that's not all to it, because the tables in this section (Tables 149 to 153) seem to be incomplete: Table 149 is missing e.g. HUGE_VALF from Table 119 [c.math], Table 150 is missing e.g. LLONG_MIN from Table 31 and DECIMAL_DIG from Table 32 [c.limits], Table 153 is missing e.g. strtoll from Table 78 [c.strings]... (sorry, I don't know how to make a list that we could be confident is complete, so I just picked one random instance in some tables).

I would be pleased if you can give me some feedback.

Thank you. Best regards

### PR DESCRIPTION
**Pull request by jensmaurer**
Consistently end the 'Changes' phrase with a period.
Add missing newlines.
Remove spurious 'Rationale' item.
Fix typos.

Fixes #214.

### CODE REVIEW

File: source/compatibility.tex [hunk 1]
```diff
@@ -159,24 +158,24 @@
...
 \difficulty
-Semantic transformation
+Semantic transformation.
 \howwide
-Seldom
+Seldom.
 
 \ref{basic.start}
-\change Main cannot be called recursively and cannot have its address taken
+\change The \tcode{main} function cannot be called recursively and cannot have its address taken.
```
Comment (tkoeppe) [inline source/compatibility.tex]: I would even delete the word "recursively". But I suppose this is non-normative, so no big deal.

File: source/compatibility.tex [hunk 2]
```diff
@@ -60,7 +60,7 @@
 Programs which depend upon \tcode{sizeof('x')} are probably rare.
 
 Subclause \ref{lex.string}:
-\change String literals made const\\
+\change String literals made const.\\
```
Comment (tkoeppe) [inline source/compatibility.tex]: Below you style `const` as code; should it be code here, too? When do you use which style?

File: source/compatibility.tex (hunk 1 above)
Comment (jensmaurer) [inline source/compatibility.tex]: I didn't want to change more than necessary to get the formatting non-awkward.

File: source/compatibility.tex (hunk 2 above)
Comment (jensmaurer) [inline source/compatibility.tex]: My patch does not change the "const" \tcode style, it just adds periods etc. (At least I hope so.) I believe we have a separate issue for const vs. \tcode.

**Pull request merged**